### PR TITLE
EOS-20356: Reset command for csm setup

### PR DESCRIPTION
**Pull request by udayan-y21**
Signed-off-by: Udayan Yaragattikar <udayan.yaragattikar@seagate.com>

# Backend

## Problem Statement

Story Ref (if any): https://jts.seagate.com/browse/EOS-20356

## Unit testing on RPM done
<pre>
Yes
</pre>
## Problem Description
<pre>
Reset command to cleanup all the db entries as well as stop the servies.
</pre>
## Solution
https://seagate-systems.atlassian.net/wiki/spaces/~700883026/pages/347734026/CSM-setup+Cleanup+and+Reset+Scope

## Unit Test Cases
Consul DB cleanup
image

ES DB CLeanup
image

Indices deleted
image

Unsupported feature entry delted for csm
image

csm_sgent.service stoped and disabled
image

Signed-off-by:

### CODE REVIEW

File: csm/conf/reset.py [hunk 1]
```diff
@@ -32,5 +38,77 @@ async def execute(self, command):
...
+        self.disable_and_stop_service()
+        self.directory_cleanup()
+        await self.db_cleanup()
+        await self._unsupported_feature_entry_cleanup()
         return Response(output=const.CSM_SETUP_PASS, rc=CSM_OPERATION_SUCESSFUL)
+
+    def disable_and_stop_service(self):
+        service_obj = Service('csm_agent.service')
+        if service_obj.is_enabled():
+            Log.info("Disable csm_sgent.service")
```
Comment (pawankumarsrivastava) [inline csm/conf/reset.py:57]: Disable --> Disabling

File: csm/conf/reset.py [hunk 2]
```diff
@@ -32,5 +38,77 @@ async def execute(self, command):
...
+            service_obj.disable()
+        if service_obj.get_state().state == 'active':
+            Log.info("Stop csm_sgent.service")
```
Comment (pawankumarsrivastava) [inline csm/conf/reset.py:60]: Stop --> Stopping

File: csm/conf/reset.py [hunk 3]
```diff
@@ -32,5 +38,77 @@ async def execute(self, command):
...
         """
-        Log.info("Not Preforming any Reset Since no Test Data is Generated by CSM.")
+        try:
+            Log.info("Loading Url into conf store.")
+            Conf.load(const.CSM_GLOBAL_INDEX, const.CSM_CONF_URL)
+            Conf.load(const.DATABASE_INDEX, const.DATABASE_CONF_URL)
+        except KvError as e:
+            Log.error(f"Configuration Loading Failed {e}")
+            raise CsmSetupError("Could Not Load Url Provided in Kv Store.")
+        self.disable_and_stop_service()
```
Comment (pawankumarsrivastava) [inline csm/conf/reset.py:48]: We should only proceed if service is stopped successfully

File: csm/conf/reset.py [hunk 4]
```diff
@@ -32,5 +38,77 @@ async def execute(self, command):
...
+        service_obj = Service('csm_agent.service')
+        if service_obj.is_enabled():
+            Log.info("Disable csm_sgent.service")
+            service_obj.disable()
+        if service_obj.get_state().state == 'active':
+            Log.info("Stop csm_sgent.service")
+            service_obj.stop()
+
+    def directory_cleanup(self):
+        Log.info("Delete files and folders")
```
Comment (pawankumarsrivastava) [inline csm/conf/reset.py:64]: Delete ---> Deleting

File: csm/cli/schema/csm_setup.json [hunk 5]
```diff
@@ -246,40 +246,10 @@
       "permissions_tag": "update",
       "args": [
         {
```
Comment (pawankumarsrivastava) [inline csm/cli/schema/csm_setup.json:248]: There is an entry for refresh_config which is not required. Remove it.

**Review by pawankumarsrivastava — CHANGES_REQUESTED** [review]
Change the base to t-main.

File: csm/conf/reset.py (hunk 1 above)
Comment (udayan-y21) [reply csm/conf/reset.py]: Modifications done

File: csm/conf/reset.py (hunk 3 above)
Comment (udayan-y21) [reply csm/conf/reset.py:48]: Modifications done

File: csm/cli/schema/csm_setup.json (hunk 5 above)
Comment (udayan-y21) [reply csm/cli/schema/csm_setup.json:248]: Modifications done

**Review by soniyamoholkar — APPROVED** [review]

**Review by pawankumarsrivastava — APPROVED** [review]

**Review by kupranay — APPROVED** [review]

**Pull request merged**